### PR TITLE
Fix/2314 tab url persists

### PR DESCRIPTION
**Pull request by kwongz**
### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable
- ~~I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

Demo (3 examples):

- Selection of 0 indexed tab (first tab) on first page load
- Selection of second tab, refresh page, second tab selection persists
- If the User shares url with tab that does not exist, reverts to 0 indexed tab (first tab)

https://github.com/user-attachments/assets/226a7838-bee9-44c8-b394-f967d79c7b86

Future notes:

Currently, we cannot accept multiple tabs with id props, since they will be sharing + updating the same URL. A story was built to demonstrate this. 

1. Should we handle this? 
2. Should we warn users not to do this?

Future PR issue:

- Multiple tabs with ID props share same URL
- Last tab selection will persist and overrule sibling tabs component

https://github.com/user-attachments/assets/56c38908-150c-4aee-ac5a-c1a46ce57a78

### CODE REVIEW

**Comment by kwongz** [comment]
fix #2314

File: packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte [hunk 1]
```diff
@@ -56,23 +56,28 @@
 	}
 
 	/** @type {import('./index.d.ts').TabsContext} */
-	const context = writable({ tabs: [], active: null });
+	const context = writable({ tabs: [], activeId: null });
```
Comment (kwongz) [inline packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte:59]: removed this property, doesn't seemed to be used anywhere

File: packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte [hunk 2]
```diff
@@ -56,23 +56,28 @@
...
 
 	onMount(() => {
 		const url = new URL(window.location.href);
 		const urlActive = url.searchParams.get(id);
 		if (urlActive) {
-			$context.active = urlActive;
```
Comment (kwongz) [inline packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte:65]: changing $context.active --> $context.activeId for consistency

File: packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte [hunk 3]
```diff
@@ -56,23 +56,28 @@
...
 	// Select the first tab by default
-	$: if (!$context.activeId && $context.tabs.length) {
+	$: if (!$context.activeId && $context.tabs.length && !id) {
 		$context.activeId = $context.tabs[0].id;
 	}
 
 	// Select the first tab when the active tab no longer exists
-	$: if (!$context.tabs.find((t) => t.id === $context.activeId)) {
+	$: if (!$context.tabs.find((t) => t.id === $context.activeId) && !id) {
 		$context.activeId = $context.tabs[0]?.id;
```
Comment (kwongz) [inline packages/ui/core-components/src/lib/unsorted/ui/Tabs/Tabs.svelte:81]: handles our initial issue of tab id not respecting url

**Comment by archiewood** [comment]
> Currently, we cannot accept multiple tabs with id props, since they will be sharing + updating the same URL. A story was built to demonstrate this.

I would consider this core functionality

We should be able to build a URL structure that uses the form

localhost:3000/page-name?tabs_id1=tab_value_x&tabs_id2=tab_value_y

**Comment by kwongz** [comment]
@archiewood  Multi-tab url id functionality 

Stories

https://github.com/user-attachments/assets/51b0774c-6fd6-4926-b179-71326cdb7ddc


Evidence Project:

https://github.com/user-attachments/assets/28d9f0b7-80b4-41fd-9916-a69803ec90a0

**Review by zachstence — APPROVED** [review]

**Pull request merged**